### DISCUSSION TITLE
Laravel 9 + laravel-backup

### DISCUSSION THREAD

**Discussion by duchymor** [spatie/flysystem-dropbox#83] [Q&A] [answered]
Hi, sorry if I miss something. I have troubles with upgrading to Laravel 9. Previously I did backups to dropbox. Now although versions I have already seems to be compatible (Laravel 9.5.1, laravel-backup 8.1.1, flysystem-dropbox 2.0.6) I got this type error:
`Spatie\Backup\BackupDestination\BackupDestination::__construct(): Argument #1 ($disk) must be of type ?Illuminate\Contracts\Filesystem\Filesystem, League\Flysystem\Filesystem given, called in /var/www/vendor/spatie/laravel-backup/
src/BackupDestination/BackupDestination.php on line 60`

**Comment by pmochine** [comment]
I have the same issue (https://github.com/spatie/flysystem-dropbox/issues/84)

**Comment by pmochine** [comment, marked as answer]
🎉 I just found a solution, because I looked up an alternative to dropbox. I found about [google drive](https://github.com/masbug/flysystem-google-drive-ext) and looked up their code for Laravel. And there I found the solution:

The whole code so dropbox is working again:
```
<?php

namespace App\Providers;

use Illuminate\Filesystem\FilesystemAdapter;
use Illuminate\Support\Facades\Storage;
use Illuminate\Support\ServiceProvider;
use League\Flysystem\Filesystem;
use Spatie\Dropbox\Client as DropboxClient;
use Spatie\FlysystemDropbox\DropboxAdapter;

class DropboxServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        Storage::extend('dropbox', function ($app, $config) {
            $client = new DropboxClient(
                $config['authorization_token']
            );

            $adapter = new DropboxAdapter($client);
            $driver = new Filesystem($adapter, ['case_sensitive' => false]);

            return new FilesystemAdapter($driver, $adapter);
        });
    }
}

```
